### ISSUE TITLE
Returning interface doesn't work

### ISSUE THREAD

**Issue by faiface** [vlang/v#354]
Hi! I played with interfaces on the playground and I found this code doesn't compile:

```
struct Dog { }

fn (d Dog) speak() string {
	return 'woof'
}

interface Speaker {
	speak() string
}

fn perform(s Speaker) {
	println(s.speak())
}

fn some_speaker() Speaker {
	return Dog{}
}

fn main() {
	dunno := some_speaker()
	perform(dunno)
}


```

But it definitely should compile, even the compiler says so.

```
You just found a bug. V can't compile this program, but it should. Please create a GitHub issue.
```

Edit: simplified the code

**Comment by Bowero** [comment]
```
is interface. field=speak run=1
is interface. field=speak run=2
pass=2 fn=`perform`
panic: test.v:12
function `speak` is private
```

This is the real error.

**Comment by AtjonTV** [comment]
Well, how to make it non-private?

I added `pub:` in the interface definition and `pub` to the function with no success:

```v
struct Dog { }

pub fn (d Dog) speak() string {
        return 'woof'
}

interface Speaker {
        pub: speak() string
}

fn perform(s Speaker) {
        println(s.speak())
}

fn some_speaker() Speaker {
        return Dog{}
}

fn main() {
        dunno := some_speaker()
        perform(dunno)
}
```

```bash
atjontv@TheLinuxBlackHole:/stmp$ v doggo.v 
is interface. field=speak run=1
is interface. field=speak run=2
pass=2 fn=`perform`
panic: doggo.v:12
function `speak` is private
```

System:
V: 0.1.5, self compiled from git commit: b846d02c
OS: Linux 5.1.15 (eOS 5.0)

(btw I am very new to V, and I dont know Go or C. I am mainly writing Kotlin)

**Comment by shenshouer** [comment]
I met the same issue

**Comment by LeeMingX** [comment]
My condition is same with @shenshouer .

Is it been solved? And I think function within interface that should be public default :)

**Comment by Viza74** [comment]
I saw there was a commit in 0.1.12 regarding interfaces, but this part still not working, I still got ```function 'blahblah' is private``` no matter what I do...

Is there a way to make interfaces work right now?

(Or any other way to do polimorphism /"attaching" unique functions to objects? Function pointers also don't work yet. What is the correct v-like way to do this now?)

**Comment by M4SSD35TRUCT10N** [comment]
Well, I did a run with `-show_c_cmd -debug` at current master (Win 10 1809) and this is what's printed out:
```
e:\Users\Enrico Lefass\Documents\tenebris\cerebro>v -os msvc -show_c_cmd -debug run bug_retest.v 

========== cl cmd line:
""C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.22.27905\bin\Hostx64\x64\cl.exe" -w /we4013 /volatile:ms /Fo"e:\Users\Enrico Lefass\Documents\tenebris\cerebro\bug_retest.tmp.c.obj" /Zi /MDd "e:\Users\Enrico Lefass\Documents\tenebris\cerebro\bug_retest.tmp.c" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\ucrt" -I "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.22.27905\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\shared" kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib /link /NOLOGO /OUT:"e:\Users\Enrico Lefass\Documents\tenebris\cerebro\bug_retest.exe" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.17763.0\ucrt\x64" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.17763.0\um\x64" /LIBPATH:"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.22.27905\lib\x64" /DEBUG:FULL"
==========

V error: Microsoft (R) C/C++-Optimierungscompiler Version 19.22.27905 fï¿½r x64

Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.



bug_retest.tmp.c

e:\Users\Enrico Lefass\Documents\tenebris\cerebro\bug_retest.tmp.c(4495): error C2079: "main__some_speaker" verwendet undefiniertes struct "Speaker"

e:\Users\Enrico Lefass\Documents\tenebris\cerebro\bug_retest.tmp.c(4497): error C2440: "return": "Dog" kann nicht in "Speaker" konvertiert werden

e:\Users\Enrico Lefass\Documents\tenebris\cerebro\bug_retest.tmp.c(4502): error C2079: "dunno" verwendet undefiniertes struct "Speaker"

e:\Users\Enrico Lefass\Documents\tenebris\cerebro\bug_retest.tmp.c(4504): error C2065: "Speaker_speak": nichtdeklarierter Bezeichner

e:\Users\Enrico Lefass\Documents\tenebris\cerebro\bug_retest.tmp.c(4502): error C2440: "Initialisierung": "Speaker" kann nicht in "int" konvertiert werden

e:\Users\Enrico Lefass\Documents\tenebris\cerebro>pause
Drcken Sie eine beliebige Taste . . . 

```

**Comment by serkonda7** [comment]
error with V is V 0.1.29 4774c89
```v
tmp/v/testy.tmp.c: In function ‘main__main’:
/tmp/v/testy.tmp.c:7922:16: error: implicit declaration of function ‘I_main__Speaker_to_Interface_main__Speaker’; did you mean ‘I_main__Dog_to_Interface_main__Speaker’? [-Werror=implicit-function-declaration]
 7922 |  main__perform(I_main__Speaker_to_Interface_main__Speaker(&dunno));
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                I_main__Dog_to_Interface_main__Speaker
/tmp/v/testy.tmp.c:7922:16: error: incompatible type for argument 1 of ‘main__perform’
 7922 |  main__perform(I_main__Speaker_to_Interface_main__Speaker(&dunno));
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                |
      |                int
/tmp/v/testy.tmp.c:7908:41: note: expected ‘main__Speaker’ {aka ‘_Interface’} but argument is of type ‘int’
 7908 | static void main__perform(main__Speaker s) {
...

```

**Comment by spytheman** [comment]
Still does not work yet in latest V 3b94a2b .

**Comment by spaceface777** [comment]
Works properly in the latest V 127503c; it was most likely fixed in #7776.
